dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+import { EuiTitle, EuiComboBox, EuiSpacer, EuiButton, EuiFieldText, EuiFlexItem, EuiFormRow, EuiIcon, EuiPanel, EuiText } from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { render as renderExplorerVis, selectExplorerVisualization } from '../../../../../../event_analytics/redux/slices/visualization_slice';
+import { AGGREGATION_OPTIONS } from '../../../../../../../../common/constants/explorer';
+import { ButtonGroupItem } from './config_button_group';
+import { visChartTypes } from '../../../../../../../../common/constants/shared';
+import { TabContext } from '../../../../../hooks';
+export const DataConfigPanelItem = ({
+  fieldOptionList,
+  visualizations,
+}: any) => {
+  const dispatch = useDispatch();
+  const { tabId } = useContext<any>(TabContext);
+  const explorerVisualizations = useSelector(selectExplorerVisualization)[tabId];
+
+  const { data } = visualizations;
+  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;
+
+  const newEntry = { label: "", aggregation: "", custom_label: "", name: "", side: "right" };
+
+  const [configList, setConfigList] = useState({
+    dimensions: [{ ...newEntry }],
+    metrics: [{ ...newEntry }],
+  });
+
+  useEffect(() => {
+    if (data.rawVizData?.dataConfig) {
+      setConfigList({
+        ...data.rawVizData?.dataConfig
```

Comment:
The question mark after `rawVizData` can be removed, as you have checked the same in the if condition. The if condition was successful, that is why you are inside this block. i.e. `dataConfig` is defined.

Reply:
removed question mark from above inner condition